odc/geo/_compress.py

```diff
@@ -19,18 +19,27 @@
 }
 
 
+def _verify_can_compress(xx: xr.DataArray):
+    """Returns error if array dimensions are not suitable for compress"""
+    if xx.ndim > 2:
+        xx = xx.squeeze()
+
+    if xx.ndim not in (2, 3):
+        raise ValueError(
+            f"Expected a 2 or 3 dimensional array; got {xx.ndim} dimensions {xx.dims}."
+        )
+
+
 def _compress_image(im: np.ndarray, driver="PNG", **opts) -> bytes:
     if im.ndim > 2:
         im = np.squeeze(im)
 
     if im.ndim == 3:
         h, w, nc = im.shape
         bands = np.transpose(im, axes=(2, 0, 1))  # Y,X,B -> B,Y,X
-    elif im.ndim == 2:
+    else:
         (h, w), nc = im.shape, 1
         bands = im.reshape(nc, h, w)
-    else:
-        raise ValueError(f"Expect 2 or 3 dimensional array got: {im.ndim}")
 
```

Comment:
unless code is on the "hot-path", i.e. gets executed millions of times in a tight loop, it's best to keep these kind of checks. Reasons why:

- communicating assumptions of the code to the reader
- so called "defensive programming" that makes code more resilient to future changes or mistakes in other parts of the existing code. Just because right now this is only reachable through another function that performs that check, this won't always stay that way.

Maybe replacing `raise ValueError` with `assert` to indicate that it's the responsibility of the caller to perform such a check, but it's still important to have something that states data shape expectations even within "internal" functions like this one.

Reply:
Thanks for the explanation, makes sense! Will put this back in

Reply:
Done - just left it as a ValueError for now as it felt cleaner that way.